File: youlexuan-manager-web/src/main/webapp/js/service/brandService.js
//商品服务

app.service('brandService',function ($http) {

    //添加数据
    this.add = function (entity) {
        return $http.post("../brand/save.do",entity);
    }

    //修改
    this.update=function(entity){
        return $http.post('../brand/update.do',entity);
    }
    //删除
    this.dele=function(selectids){
        return $http.post('../brand/del.do',selectids);
    }

    //获取指定id品牌信息
    this.findOne=function(entity){
        return $http.post('../brand/findOne.do',entity);
    }

    //分页条件查询
    this.search  = function (pageNumber,pageSize,criteria) {

       return $http.post("../brand/findPageCriteria.do?pageNumber="+pageNumber+"&pageSize="+pageSize,criteria);
    }

    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }

})

File: youlexuan-sellergoods-interface/src/main/java/com/zl/sellergoods/service/BrandService.java
package com.zl.sellergoods.service;

import com.zl.entity.PageResult;
import com.zl.entity.Result;
import com.zl.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    public List<TbBrand> findAll();

    public PageResult findPage(int pageNumber,int pageSize);

    public Result addBrand(TbBrand brand);

    public TbBrand findOne(long id);

    public Result update(TbBrand brand);

    public Result delete(Long[] ids);

    public PageResult findPage(TbBrand brand,int pageNumber,int pageSize);

    public List<Map> selectOptionList();
}
